packages/aws-api/src/listAllResources/listAllResourcesFromCloudformation.ts

```diff
@@ -74,33 +76,89 @@ const createARNFromCloudFormation = ({
   }
 };
 
+const findCloudformationStacksToSearch = async (
+  cloudformationStacksToFilter: string[],
+  cloudFormationClient: CloudFormationClient,
+) => {
+  const cloudformationStacks: string[] = [];
+  if (cloudformationStacksToFilter.length > 0) {
+    return cloudformationStacksToFilter;
+  }
+  for await (const page of paginateListStacks(
+    {
+      client: cloudFormationClient,
+    },
+    {
+      StackStatusFilter: Object.values(StackStatus).filter(
+        status =>
+          status !== StackStatus.DELETE_COMPLETE &&
+          status !== StackStatus.DELETE_FAILED &&
+          status !== StackStatus.DELETE_IN_PROGRESS,
+      ),
+    },
+  )) {
+    cloudformationStacks.push(
+      ...(page.StackSummaries?.map(stack => stack.StackName ?? '') ?? []),
+    );
+  }
+
+  return cloudformationStacks;
+};
+
+type ResourcesWithStackName = StackResourceSummary & { StackName: string };
+
 export const listAllResourcesFromCloudformation = async (
-  cloudformationStacks: string[],
-): Promise<CustomARN[]> => {
+  cloudformationStacksToFilter: string[],
+): Promise<{ arn: CustomARN; stackName: string }[]> => {
   const cloudFormationClient = new CloudFormationClient({});
 
-  const resources: StackResourceSummary[] = [];
-  for (const stack of cloudformationStacks) {
-    for await (const page of paginateListStackResources(
-      { client: cloudFormationClient },
-      { StackName: stack },
-    )) {
-      resources.push(...(page.StackResourceSummaries ?? []));
+  const cloudformationStacksToSearch = await findCloudformationStacksToSearch(
+    cloudformationStacksToFilter,
+    cloudFormationClient,
+  );
+
+  const resources: ResourcesWithStackName[] = [];
+  for (const stack of cloudformationStacksToSearch) {
+    try {
+      for await (const page of paginateListStackResources(
+        { client: cloudFormationClient },
+        { StackName: stack },
+      )) {
+        page.StackResourceSummaries?.forEach(resource =>
+          resources.push({ ...resource, StackName: stack }),
```

Comment:
could be cool (at least for the V1) to reference the root stack instead of the current stack when the stack is a nested

you have access to RootId and StackId when you list stacks line 101, this way you could build a stackName -> rootStackName mapping and use it here